src/index.ts

```diff
@@ -349,7 +349,221 @@ class DefaultVisualizer {
+    private _pinned: GUID[];
+
+    // the constructor is private because the main entry point is the static
+    // `load` function
+    private constructor(config: StructureConfig, dataset: Dataset) {
+        // validateConfig(config as unknown as JsObject); // validate for the momentdon't
+        validateDataset(dataset as unknown as JsObject);
+
+        this._dataset = dataset;
+        this._pinned = [];
+
+        const mode = dataset.environments === undefined ? 'structure' : 'atom';
+        this._indexer = new EnvironmentIndexer(mode, dataset.structures, dataset.environments);
+
+        this.meta = new MetadataPanel(config.meta, dataset.meta);
+
+        // Structure viewer setup
+        this.structure = new ViewersGrid(
+            config.structure,
+            this._indexer,
+            dataset.structures,
+            dataset.environments,
+            1 // hardcoded single panel
+        );
+
+        if (config.loadStructure !== undefined) {
+            this.structure.loadStructure = config.loadStructure;
+        }
+
+        this.structure.activeChanged = (guid, indexes) => {
+            this.info.show(indexes);
+        };
+
+        this.structure.onselect = (indexes) => {
+            this.info.show(indexes);
+        };
+
+        this.structure.onremove = (guid) => {
+            // remove the guid from this._pinned
+            const index = this._pinned.indexOf(guid);
+            assert(index > -1);
+            this._pinned.splice(index, 1);
+        };
+
+        this.structure.oncreate = (guid, color, indexes) => {
+            this.info.show(indexes);
+            this._pinned.push(guid);
+        };
+
+        // information table & slider setup
+        this.info = new EnvironmentInfo(config.info, dataset.properties, this._indexer);
+        this.info.onchange = (indexes) => {
+            this.structure.show(indexes);
+        };
+        this.info.startStructurePlayback = (advance) => this.structure.structurePlayback(advance);
+        this.info.startAtomPlayback = (advance) => this.structure.atomPlayback(advance);
+
+        let initial: Indexes = { environment: 0, structure: 0, atom: 0 };
+        if (config.settings && config.settings.pinned) {
+            initial = this._indexer.from_environment(config.settings.pinned[0]);
+        }
+
+        const firstGUID = this.structure.active;
+        this._pinned.push(firstGUID);
+        this.structure.show(initial);
+        this.info.show(initial);
+
+        // setup additional pinned values from the settings if needed
+        if (config.settings !== undefined) {
+            delete config.settings.map;
+            this.applySettings(config.settings);
+        }
+    }
+
+    /**
+     * Removes all the chemiscope widgets from the DOM
+     */
+    public remove(): void {
+        this.meta.remove();
+        this.info.remove();
+        this.structure.remove();
+    }
+
+    /**
```

Comment:
I don't think we want to have the `saveSettings`, `applySettings`, or `dataset` for StructureVisualizer. They are purely duplicated code which no-one uses, and would be relatively easy to add back if we need them later.

Reply:
I agree for the jupyter use case but how about the object that can be used in a webpage? Don't we want to give the possibility of applying settings? Alternative would be to move these to the chemiscope app and out of the viewers.

Reply:
> how about the object that can be used in a webpage?

Since nobody is using this functionality, I would use YAGNI and remove code. 

Moving the settings functions to the app is also a good idea, I'll do it in a separate PR once this one is merged.